domain/src/main/java/com/onboarding/domain/usecase/GetAllCharactersUseCaseImpl.kt

```diff
@@ -0,0 +1,29 @@
+package com.onboarding.domain.usecase
+
+import com.onboarding.domain.entity.CharacterResult
+import com.onboarding.domain.util.Result
+
+interface GetAllCharactersUseCase {
+    fun getAllCharacters(): Result<List<CharacterResult>>
+}
+
+class GetAllCharactersUseCaseImpl(private val service: RickAndMortyRepository, private val database: RickAndMortyDatabase) :
+    GetAllCharactersUseCase {
+
+    override fun getAllCharacters(): Result<List<CharacterResult>> =
+        when (val baseResponse = service.getAllCharacters()) {
+            is Result.Success -> {
+                val charactersList = baseResponse.data.results
+                database.insertCharacters(charactersList)
+                Result.Success(charactersList)
```

Comment:
here you should take the info from the database

Reply:
used database